crates/web5/src/crypto/key_managers/in_memory_key_manager.rs

```diff
@@ -31,31 +34,166 @@ impl InMemoryKeyManager {
-                "Provided JWK does is not a private key".to_string(),
+            return Err(Web5Error::Parameter(
+                "private_jwk must be a private key".to_string(),
             ));
         }
 
         let mut public_jwk = private_jwk.clone();
         public_jwk.d = None;
 
-        let mut map_lock = self.map.write().map_err(|e| {
-            KeyManagerError::InternalKeyStoreError(format!("unable to acquire Mutex lock: {}", e))
-        })?;
+        let mut map_lock = self.map.write()?;
         map_lock.insert(public_jwk.compute_thumbprint()?, private_jwk);
         Ok(public_jwk)
     }
 }
 
 impl KeyManager for InMemoryKeyManager {
     fn get_signer(&self, public_jwk: Jwk) -> Result<Arc<dyn Signer>> {
-        let map_lock = self.map.read().map_err(|e| {
-            KeyManagerError::InternalKeyStoreError(format!("Unable to acquire Mutex lock: {}", e))
-        })?;
+        if !public_jwk.is_public_key() {
+            return Err(Web5Error::Parameter(
+                "public_jwk must be a public key".to_string(),
+            ));
+        }
+
+        let map_lock = self.map.read()?;
         let thumbprint = public_jwk.compute_thumbprint()?;
         let private_jwk = map_lock
             .get(&thumbprint)
-            .ok_or(KeyManagerError::KeyNotFound(thumbprint))?;
+            .ok_or(Web5Error::NotFound(format!(
+                "signer not found for public_jwk with thumbprint {}",
+                thumbprint
+            )))?;
         Ok(Arc::new(Ed25519Signer::new(private_jwk.clone())))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::crypto::dsa::ed25519::Ed25519Generator;
+    use crate::{test_helpers::UnitTestSuite, test_name};
+    use std::sync::LazyLock;
+
+    mod import_private_jwk {
+        use super::*;
+
+        static TEST_SUITE: LazyLock<UnitTestSuite> =
+            LazyLock::new(|| UnitTestSuite::new("in_memory_key_manager_import_private_jwk"));
+
+        #[test]
+        fn z_assert_all_suite_cases_covered() {
+            // fn name prefixed with `z_*` b/c rust test harness executes in alphabetical order,
+            // unless intentionally executed with "shuffle" https://doc.rust-lang.org/rustc/tests/index.html#--shuffle
+            // this may not work if shuffled or if test list grows to the extent of 100ms being insufficient wait time
+
+            // wait 100ms to be last-in-queue of mutex lock
+            std::thread::sleep(std::time::Duration::from_millis(100));
+
+            TEST_SUITE.assert_coverage()
+        }
+
+        #[test]
+        fn test_must_be_private_jwk() {
+            TEST_SUITE.include(test_name!());
+
+            let key_manager = InMemoryKeyManager::new();
+            let private_jwk = Ed25519Generator::generate();
+            let mut public_jwk = private_jwk.clone();
+            public_jwk.d = None;
+
+            let result = key_manager.import_private_jwk(public_jwk);
+            assert!(result.is_err());
```

Comment:
nit: instead of asserting `is_err()` and `is_ok()`, how do you feel about `expect_err("private_jwk must be a private key")` and `expect("whatever the helpful error message is")`? slightly tighter than asserting Ok/Err than unwrapping in the following line